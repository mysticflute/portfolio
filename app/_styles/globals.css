/* The file contains root-level custom properties and element resets only. */

:root {
  /* colors */
  --color-gray-100: #0b0b0b;
  --color-gray-200: #393939;
  --color-gray-300: #5e5e5f;
  --color-gray-400: #878787;
  --color-neutral-100: #fff;
  --color-neutral-200: #eff0f6;
  --color-neutral-300: #eceae8;
  --color-red-100: #e85d75;
  /* --color-red-100: #e574bc; */
  --color-yellow-100: #f6b468;
  --color-blue-100: #53abf8;
  --color-purple-100: #4331f0;
  --color-purple-200: #5358f8;
  --color-primary: var(--color-purple-100);
  --color-accent-100: var(--color-purple-200);
  --color-accent-200: var(--color-blue-100);
  --color-accent-300: var(--color-yellow-100);
  --color-accent-400: var(--color-red-100);
  --color-background-100: var(--color-gray-100);
  --color-background-200: var(--color-neutral-100);
  --color-background-300: var(--color-gray-100);
  --color-background-highlight: var(--color-blue-100);
  --color-text: var(--color-gray-200);
  --color-text-accent: var(--color-gray-100);
  --color-text-alt: var(--color-gray-100);
  --color-text-link: var(--color-gray-100);
  --color-text-highlight: var(--color-blue-100);
  --color-text-inverse: var(--color-neutral-200);

  /* spacing */
  --layout-max-width: 1300px;
  --layout-padding-x: 24px;
  --layout-padding-y: 0;
  --layout-padding: var(--layout-padding-y) var(--layout-padding-x);
  --layout-margin-x: auto;
  --layout-margin-y: 0;
  --layout-margin: var(--layout-margin-y) var(--layout-margin-x);
}

*,
::before,
::after {
  box-sizing: border-box;
}

html {
  scroll-behavior: smooth;
  scroll-padding: 2rem;
}

body {
  color: var(--color-text);
  background: var(--color-primary);
  font-family: var(--font-default), system-ui, sans-serif;
  font-weight: 500;
  font-size: 1.125rem;
  line-height: 1.667em;
}

@media screen and (max-width: 767px) {
  body {
    font-size: 1rem;
  }
}

h1,
h2,
h3,
h4,
h5,
h6 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  font-size: 1rem;
}

ol,
ul {
  margin: 0;
  padding: 0;
  list-style: none;
}

p {
  margin: 0;
  padding: 0;
}

audio,
canvas,
iframe,
img,
svg,
video {
  vertical-align: middle;
}

img {
  max-width: 100%;
  height: auto;
}

a {
  color: inherit;
  text-decoration: none;
}

@media screen and (max-width: 991px) {
  :root {
    --layout-max-width: calc(600px + var(--layout-padding-x) * 2);
  }
}

@media screen and (max-width: 767px) {
  :root {
    --layout-max-width: calc(500px + var(--layout-padding-x) * 2);
  }
}

@media screen and (max-width: 479px) {
  :root {
    --layout-padding-x: 20px;
  }
}
