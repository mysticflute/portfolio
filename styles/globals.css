:root {
  --color-dark: #0b0b0b;
  --color-blue: #1c92ff;
  --color-red: #ff4a60;
  --color-yellow: #ffc431;
  --color-text-default: #393939;
  --color-text-dark: var(--color-dark);
  --color-text-header-link: var(--color-dark);
  --color-text-header-link-highlight: var(--color-blue);
  --color-background-default: #faf9f6;
  --color-background-dark: var(--color-dark);
  --color-background-highlight: var(--color-blue);
}

*,
::before,
::after {
  box-sizing: border-box;
}

body {
  color: var(--color-text-default);
  background: var(--color-background-default);
  font-family: var(--font-default), system-ui, sans-serif;
  font-weight: 500;
  font-size: 1.125rem;
  line-height: 1.667em;
}

@media screen and (max-width: 767px) {
  body {
    font-size: 1rem;
  }
}

h1,
h2,
h3,
h4,
h5,
h6 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  font-size: 1rem;
}

ol,
ul {
  margin: 0;
  padding: 0;
  list-style: none;
}

p {
  margin: 0;
  padding: 0;
}

audio,
canvas,
iframe,
img,
svg,
video {
  vertical-align: middle;
}

img {
  max-width: 100%;
  height: auto;
}

a {
  color: inherit;
  text-decoration: none;
}

/* @media (prefers-color-scheme: dark) {
  html {
    color-scheme: dark;
  }
} */

/* @media (prefers-color-scheme: dark) {
  :root {
  }
} */
