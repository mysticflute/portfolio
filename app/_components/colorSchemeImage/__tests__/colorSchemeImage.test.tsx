/**
 * @jest-environment jsdom
 */

import { describe, expect, it } from '@jest/globals';
import { render } from '@testing-library/react';
import ColorSchemeImage from '../colorSchemeImage';

const srcLight = {
  src: '/srcLight.png',
  height: 1,
  width: 1,
};

const srcDark = {
  src: '/srcDark.png',
  height: 1,
  width: 1,
};

describe('colorSchemeImage', () => {
  it('matches the snapshot', () => {
    const { container } = render(
      <ColorSchemeImage srcLight={srcLight} srcDark={srcDark} alt="alt text" />,
    );

    expect(container).toMatchSnapshot();
  });
});
