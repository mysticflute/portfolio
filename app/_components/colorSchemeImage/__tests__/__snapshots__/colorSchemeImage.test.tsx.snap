// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`colorSchemeImage matches the snapshot with additional props 1`] = `
<div>
  <picture>
    <source
      media="(prefers-color-scheme: dark)"
      sizes="(max-width: 500px) 100vw, 50vw"
      srcset="/_next/image?url=%2FsrcDark.png&w=384&q=80 384w, /_next/image?url=%2FsrcDark.png&w=640&q=80 640w, /_next/image?url=%2FsrcDark.png&w=750&q=80 750w, /_next/image?url=%2FsrcDark.png&w=828&q=80 828w, /_next/image?url=%2FsrcDark.png&w=1080&q=80 1080w, /_next/image?url=%2FsrcDark.png&w=1200&q=80 1200w, /_next/image?url=%2FsrcDark.png&w=1920&q=80 1920w, /_next/image?url=%2FsrcDark.png&w=2048&q=80 2048w, /_next/image?url=%2FsrcDark.png&w=3840&q=80 3840w"
    />
    <img
      alt="alt text"
      decoding="async"
      fetchpriority="high"
      height="100"
      sizes="(max-width: 500px) 100vw, 50vw"
      src="/_next/image?url=%2FsrcLight.png&w=3840&q=80"
      srcset="/_next/image?url=%2FsrcLight.png&w=384&q=80 384w, /_next/image?url=%2FsrcLight.png&w=640&q=80 640w, /_next/image?url=%2FsrcLight.png&w=750&q=80 750w, /_next/image?url=%2FsrcLight.png&w=828&q=80 828w, /_next/image?url=%2FsrcLight.png&w=1080&q=80 1080w, /_next/image?url=%2FsrcLight.png&w=1200&q=80 1200w, /_next/image?url=%2FsrcLight.png&w=1920&q=80 1920w, /_next/image?url=%2FsrcLight.png&w=2048&q=80 2048w, /_next/image?url=%2FsrcLight.png&w=3840&q=80 3840w"
      style="color: transparent;"
      width="200"
    />
  </picture>
</div>
`;

exports[`colorSchemeImage matches the snapshot with basic props 1`] = `
<div>
  <picture>
    <source
      media="(prefers-color-scheme: dark)"
      srcset="/_next/image?url=%2FsrcDark.png&w=256&q=75 1x, /_next/image?url=%2FsrcDark.png&w=640&q=75 2x"
    />
    <img
      alt="alt text"
      decoding="async"
      height="100"
      loading="lazy"
      src="/_next/image?url=%2FsrcLight.png&w=640&q=75"
      srcset="/_next/image?url=%2FsrcLight.png&w=256&q=75 1x, /_next/image?url=%2FsrcLight.png&w=640&q=75 2x"
      style="color: transparent;"
      width="200"
    />
  </picture>
</div>
`;
